File: android/src/main/java/com/audiocontext/nodes/GainNode.kt
package com.audiocontext.nodes

import com.audiocontext.context.BaseAudioContext
import com.facebook.jni.HybridData

class GainNode(context: BaseAudioContext): AudioNode(context) {
  override val numberOfInputs: Int = 1
  override val numberOfOutputs: Int = 1
  private var gain: Double = 1.0
    get() = field
    set(value) {
      field = value
    }

  private val mHybridData: HybridData? = initHybrid();

  override fun process(playbackParameters: PlaybackParameters) {
    playbackParameters.gain = gain
    super.process(playbackParameters)
  }
}

File: packages/react-native-audio-api/src/mobile/effects/GainNode.ts
import { IGainNode } from '../interfaces';
import AudioNode from '../core/AudioNode';
import AudioParam from '../core/AudioParam';
import BaseAudioContext from '../core/BaseAudioContext';

export default class GainNode extends AudioNode {
  readonly gain: AudioParam;

  constructor(context: BaseAudioContext, gain: IGainNode) {
    super(context, gain);
    this.gain = new AudioParam(gain.gain, context);
  }
}
